packages/engine/Source/Scene/VoxelCell.js

```diff
@@ -0,0 +1,268 @@
+import Cartesian3 from "../Core/Cartesian3.js";
+import Check from "../Core/Check.js";
+import defined from "../Core/defined.js";
+import MetadataType from "./MetadataType.js";
+import OrientedBoundingBox from "../Core/OrientedBoundingBox.js";
+
+/**
+ * A cell from a {@link VoxelPrimitive}.
+ * <p>
+ * Provides access to properties associated with one cell of a voxel primitive.
+ * </p>
+ * <p>
+ * Do not construct this directly.  Access it through picking using {@link Scene#pickVoxel}.
+ * </p>
+ *
+ * @alias VoxelCell
+ * @constructor
+ *
+ * @param {VoxelPrimitive} primitive The voxel primitive containing the cell
+ * @param {number} tileIndex The index of the tile
+ * @param {number} sampleIndex The index of the sample within the tile, containing metadata for this cell
+ *
+ * @example
+ * // On left click, display all the properties for a voxel cell in the console log.
+ * handler.setInputAction(function(movement) {
+ *   const voxelCell = scene.pickVoxel(movement.position);
+ *   if (voxelCell instanceof Cesium.VoxelCell) {
+ *     const propertyIds = voxelCell.getPropertyIds();
+ *     const length = propertyIds.length;
+ *     for (let i = 0; i < length; ++i) {
+ *       const propertyId = propertyIds[i];
+ *       console.log(`{propertyId}: ${voxelCell.getProperty(propertyId)}`);
+ *     }
+ *   }
+ * }, Cesium.ScreenSpaceEventType.LEFT_CLICK);
+ */
+function VoxelCell(primitive, tileIndex, sampleIndex) {
+  this._primitive = primitive;
+  this._tileIndex = tileIndex;
+  this._sampleIndex = sampleIndex;
+  this._metadata = {};
+  this._orientedBoundingBox = new OrientedBoundingBox();
+}
+
+/**
+ * Construct a VoxelCell, and update the metadata and bounding box using the properties
+ * of a supplied keyframe node.
+ *
+ * @private
+ * @param {VoxelPrimitive} primitive The voxel primitive containing the cell.
+ * @param {number} tileIndex The index of the tile.
+ * @param {number} sampleIndex The index of the sample within the tile, containing metadata for this cell.
+ * @param {KeyframeNode} keyframeNode The keyframe node containing information about the tile.
+ * @returns {VoxelCell}
+ */
+VoxelCell.fromKeyframeNode = function (
+  primitive,
+  tileIndex,
+  sampleIndex,
+  keyframeNode
+) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.object("primitive", primitive);
+  Check.typeOf.number("sampleIndex", sampleIndex);
```

Comment:
Should `tileIndex` be here?

Reply:
Yes, for consistency. I checked the others because they are needed for subsequent function calls, whereas tileIndex is only reported back to the user. But we might as well check all of them.